src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,234 @@
+            parameters = retrieveParameters(input);
+        }
+        exit(0);
+    }
+
+    public static void executeCommand(String command, String parameters) {
+        String deadline;
+        String description;
+        int taskNumber;
+
+        switch (command) {
+        case "help":
+            taskManager.printHelpMessage();
+            break;
+        case "list":
+            taskManager.listTasks();
+            break;
+        case "mark":
+            taskNumber = retrieveTaskNumber(parameters);
+            taskManager.markTaskAsDone(taskNumber);
+            fileDataManager.saveData(taskManager);
+            break;
+        case "unmark":
+            taskNumber = retrieveTaskNumber(parameters);
+            taskManager.markTaskAsUndone(taskNumber);
+            fileDataManager.saveData(taskManager);
+            break;
+        case "delete":
+            taskNumber = retrieveTaskNumber(parameters);
+            taskManager.removeTask(taskNumber);
+            fileDataManager.saveData(taskManager);
+            break;
+        case "todo":
+            taskManager.addTask(new Todo(parameters));
+            fileDataManager.saveData(taskManager);
+            break;
+        case "deadline":
+            description = retrieveTaskDescription(parameters, TaskManager.DEADLINE_SEPERATOR);
+            deadline = retrieveTime(parameters, TaskManager.DEADLINE_SEPERATOR);
+            taskManager.addTask(new Deadline(description, deadline));
+            fileDataManager.saveData(taskManager);
+            break;
+        case "event":
+            description = retrieveTaskDescription(parameters, TaskManager.EVENT_SEPERATOR);
+            deadline = retrieveTime(parameters, TaskManager.EVENT_SEPERATOR);
+            taskManager.addTask(new Event(description, deadline));
+            fileDataManager.saveData(taskManager);
+            break;
+        }
+    }
+
+    public static void validateFormat(String command, String parameters) throws IncorrectFormatException, NumberFormatException {
+        if (command.equals("list") || command.equals("help"))
+            return;
+        if (command.equals("mark") || command.equals("unmark") || command.equals("delete")) {
+            if (parameters.isEmpty()) {
+                throw new IncorrectFormatException("Incorrect format! The task number that is to be marked as done/not done cannot be empty!");
+            }
+            retrieveTaskNumber(parameters);
+        }
+        if (command.equals("deadline")) {
+            handlePossibleDeadlineExceptions(parameters);
+        }
+        if (command.equals("event")) {
+            handlePossibleEventExceptions(parameters);
+        }
+        if (parameters.isEmpty()) {
+            throw new IncorrectFormatException("Incorrect format! The description of a todo cannot be empty!");
+        }
+    }
+
+    public static void handlePossibleEventExceptions(String parameters) throws IncorrectFormatException {
```

Comment:
A lot of lines in this and the next function are >120 characters. Do consider changing them to multi-line or shorten them.